### ISSUE TITLE
Question about library version

### ISSUE THREAD

**Issue by byminji** [muzairkhattak/ViFi-CLIP#10]
Hi, thanks for sharing nice work.

While reproducing the HMDB51 16-shot setting, I got pid killed error like this:

<img width="1070" alt="Screenshot 2023-10-18 at 3 14 06 PM" src="https://github.com/muzairkhattak/ViFi-CLIP/assets/65417836/44b2bd50-5fd9-49d5-a103-ce9af47e7037">

![Screenshot 2023-10-18 at 10 10 46 PM](https://github.com/muzairkhattak/ViFi-CLIP/assets/65417836/7173ce2e-7dee-4687-b82a-6d728394de5c)

When I set `num_workers=0` and `pin_memory=False` in train data loader, this kind of error is resolved. I'm guessing this might be related to video decoding process of mmcv-full or decord. Could you please share the full version of the library in `requirements.txt` in your environment?

Thanks :)

**Comment by muzairkhattak** [comment]
Hi @byminji,

Thank you for your message !

Sorry for responding late to you, we were digging the details about the packages. I have shared the details of all packages with their exact version [in this text file](https://drive.google.com/file/d/1zrw-ctsMu9P45FwLYa4NER7CIZhdsWS2/view?usp=sharing). You can refer to this.

I hope that will be helpful.

Thank you and kind regards!

**Comment by byminji** [comment]
Hi @muzairkhattak , thank you for kindly sharing all package versions! Based on the text file and README, I've tried reproducing the results in the paper with V100 GPUs, but some settings appear to result in a 1-3% drop in accuracy compared to the paper. When I try the fully-supervised and zero-shot settings, the accuracy is reproduced. However, I failed to reproduce some results of few-shot VL prompting, base2novel fine-tuning, and base2novel VL prompting. Regarding this, I have several additional questions about the implementation details:

1. During few-shot and base2novel training, did you use the best checkpoint or the least checkpoint?
2. I also tried using both 8 GPUs and 4 GPUs with the same effective batch size by adjusting accumulation steps, and it seems that the results are not consistent when I change the accumulation steps. Did you also experience this kind of behavior?
3. Could you please share the training logs if they're available?

Thank you so much for your kind regards. :)

**Comment by muzairkhattak** [comment]
Hi @byminji,

Thank you for the update !

Regarding your queries, please note the following:
1. Our work is based on XCLIP and its repository. Following this prior work, we used the best checkpoint results for fair comparisons. 
2. Actually we have never verified this behavior. Ideally it should be similar. Can you specify how large are the inconsistencies? 
3. Unfortunately we do not have any training logs stored anywhere.

Lastly, we have experienced fluctuations in reproducing few-shot results when tested on different machines. So I think +/- 1-2 percent is common in few-shot setting as training results variance is usually large. 

Thank you very much !

**Comment by byminji** [comment]
Hi @muzairkhattak , thank you for your quick and kind reply :)

About the inconsistency in using 4GPUs and 8GPUs, the maximum gap was:
- zero-shot setting: marginal difference (0.1%p) in HMDB51 and K600 / 0.5%p drop in 4GPU in UCF 101
- few-shot and base2novel fine-tuning: 0.2~0.7%p gap
- few-shot and base2novel VL prompting: 1~2%p gap, and up to 3%p for novel class

In most settings using 8GPUs are better than 4GPUs, and otherwise the opposite. I think the difference in fully-supervised and zero-shot settings are marginal, but in few-shot and base2novel settings the gap is larger. Is it related to some data loading differences by changing the number of gpus, even though we fix the seed?

**Comment by muzairkhattak** [comment]
Hi @byminji,

I think using different configuration of GPUs could have 1-2 % variation in the few-shot setting scenario. This can be considered as common. 

Actually, for few-shots, the data samples are very less and the learning curve of the model is very noisy, so for each run, there is more chance of fluctuations. 

Finally for your case, you can stick with 8 GPUs if that is providing better results. 
I hope that could be helpful.

Thank you and kind regards!

Muhammad Uzair
